torch/nn/parallel/distributed.py

```diff
@@ -661,6 +684,20 @@ def __init__(
             os.environ.get("PYTORCH_DDP_USE_SIDE_STREAM", "1") == "1"
         )
 
+        # Initialize gradient buffers and register all reduce hook
+        self._grad_buffer = None
+        self._grad_views = []
+        self._use_vanilla_module = False
```

Comment:
nit: self._delay_all_reduce = False

Reply:
The flag "use_vanilla_module" is used to flag whether all parameters are specified in delay_all_reduce_named_params. In this case, we don't need DDP and use the forward function of the self.module.